### ISSUE TITLE
Crash on slice

### ISSUE THREAD

**Issue by sdooweloc** [supermerill/Slic3r#2960]
**Bug description**
When slicing a model the application crashes. Tried multiple different profiles but to no avail.

**Expected behavior**
For the model to properly slice and present me with useable GCode

!!>> Project File <<!!

[300mmLeftFixedKlickyBucket.zip](https://github.com/supermerill/SuperSlicer/files/9149637/300mmLeftFixedKlickyBucket.zip)

**To Reproduce**
Load in STL and hit slice 

**Desktop (please complete the following information):**
 - OS: Arch Linux Kernel 5.18.11-arch1-1 with KDE Plasma 5.25.3
 - Version: 2.4.58.2
Build: SuperSlicer_2.4.58.2+UNKNOWN
 - Printer model: Voron 2.4

**Additional context**
Logging: https://pastebin.com/uqrfmhD2

**Comment by tome9111991** [comment]
i dont have issue with your file on:
Manjaro Linux
is use latest Nightly from susi

**Comment by neophyl** [comment]
Might not be useful for you immediately but it rules your file out and is a data point for Merill who potentially has to fix the issue.

**Comment by supermerill** [comment]
can't replicate with either 2.4.2 nor 2.4.3 on ubuntu 20.04

**Comment by sdooweloc** [comment]
Not really sure how to rule out bad version. None of the app images will run, I cant run the latest from the arch repos due to another bug open, this I had to forcibly downgrade it and other packages just to get it to run at all on my machine due to that other bug causing crashes on launch after update.

**Comment by Wolf220** [comment]
I'm actually having the same issue, (same distro, different de) but it doesn't crash when I slice with a brim.

**Comment by supermerill** [comment]
Seems like there is a sketchy random bug. I'll have to do some extensive dangerous code hunting.

**Comment by raku-cat** [comment]
I'm also on Arch experiencing this issue.
The previously mentioned bug has been since fixed: https://bugs.archlinux.org/index.php?do=details&action=details.addvote&task_id=75384
So everything can be ran at an "up to date" version from a package management perspective.
This occurs for me mostly with "more complex" models in my experience, most things slice fine but in particular vase mode has given me trouble, as well as more intricate models if I don't scale them down or rotate them to face a certain direction.

This model, for example: https://www.thingiverse.com/thing:5436449/files
I cannot slice in vase mode, it makes some progress, the program locks up, and then crashes with the following:
```
superslicer: /build/superslicer/src/SuperSlicer/src/slic3r/GUI/3DScene.cpp:263: void Slic3r::GLIndexedVertexArray::render(const std::pair<long unsigned int, long unsigned int>&, const std::pair<long unsigned int, long unsigned int>&) const: Assertion `this->vertices_and_normals_interleaved_VBO_id != 0' failed.
zsh: IOT instruction (core dumped)  superslicer
```

I'm running 2.4.58.3 on Arch which is the latest we have available from the official repos: https://archlinux.org/packages/community/x86_64/superslicer/
Let me know of anything else I can provide!

**Comment by supermerill** [comment]
@raku-cat 
> Edit: I've tried the brim fix an above commenter mentioned, the slicing does get farther (Shows the actual gcode preview, and the status shows "generating gcode") before crashing, and there's a different error, as well:

can you share the project file (file -> save project as) ?

**Comment by supermerill** [comment]
> I'm assuming you mean do this on a plate that would crash, prior to me generating g-code and crashing the program.

If when you load it, and slice, it crash, it's perfect.

**Comment by raku-cat** [comment]
> > I'm assuming you mean do this on a plate that would crash, prior to me generating g-code and crashing the program.
> 
> If when you load it, and slice, it crash, it's perfect.

Yep, the one I linked crashes on slice, with or without vase mode set.
On additional testing, I can get it to slice without crashing by scaling it down to about 80% or lower.

**Comment by supermerill** [comment]
can't make it crashs on ubuntu on a recent version

**Comment by jaimebpg** [comment]
I am also having crashes when slicing on arch linux when I resize the object or use XY compensation.

Version: 2.4.58.3
Build: SuperSlicer_2.4.58.3+UNKNOWN

superslicer: /build/superslicer/src/SuperSlicer/src/libslic3r/GCode.cpp:3933: std::string Slic3r::GCode::extrude_loop(const Slic3r::ExtrusionLoop&, const std::string&, double, std::unique_ptr<Slic3r::EdgeGrid::Grid>*): Assertion `false' failed.
Abortado (`core' generado)

**Comment by supermerill** [comment]
@jaimebpg  That one detect an unconnected loop. Can you share your project (I guess you're on the 2.4.58.3 release)?

I wonder what's different in arch.

**Comment by jaimebpg** [comment]
> @jaimebpg That one detect an unconnected loop. Can you share your project (I guess you're on the 2.4.58.3 release)?
> 
> I wonder what's different in arch.

https://www.dropbox.com/s/sd7mhnz7wckh06o/1_Transmission_V3.6.3mf?dl=0

**Comment by supermerill** [comment]
@jaimebpg can reproduce on 2.4.58.3, but not on the nightly, so I guess it's already fixed for next release

**Comment by raku-cat** [comment]
I updated to 2.4.58.4 by manually upping the package version on arch since the repo hasn't/hadn't updated yet, definitely more stable for most instances, I can't actually slice a benchy anymore though ;) Error is as follows:
```
superslicer: /data/tmp/superslicer/src/SuperSlicer/src/libslic3r/GCode.cpp:3942: std::string Slic3r::GCode::extrude_loop(const Slic3r::ExtrusionLoop&, const std::string&, double, std::unique_ptr<Slic3r::EdgeGrid::Grid>*): Assertion `false' failed.
zsh: IOT instruction (core dumped)  superslicer
```

**Comment by top4ek** [comment]
SS 2.4.58.5, Archlinux Linux 5.19.13-arch1-1
IOT instruction (core dumped)  superslicer

**Comment by eduncan911** [comment]
@supermerill as you can see in the issues, there are many many segment faults/core dumps on Linux with the Oct 2022 version (me included).  Multiple installs and even different machines.

People say when this or that is checked, when doing this or that file, etc.  After ~1800 hours on my printers since Oct 2022, with exactly this problem on this version across machines, like 1 out of 5 slices, I can tell you: **various settings and STLs do not matter.** 

I've been trying to capture outputs and logs with `--loglevel=5` for several months.  However, nothing is common among the logging - and generally logging doesn't show diddly but normal debug/trace logging on different parts.

Something is fluxed, and it's different per machine (desktop is pop_os 2204, laptop is pop_os 2204, kids' machine is arch linux (latest).  They all crash, on different STLs, on different machines, different printers, with different settings, at different projects.  There really is no commonality to any of these.

One machine crashes, I take the exact same project file to an identical updated/patched machine with the exact same software/kernel/etc.  And it slices fine, all day (e.g. I had to slice the same STL with a slight tweak, 22 times... on the kid's gaming machine because that same STL crashed 100% the time on my desktop).

---

One work around that works about ~75% of the time is to just slightly move one of the parts like 1 or 2mm.  Then it slices just fine.  

I also found if parts are too close, it will crash 100% of the time regardless of moving 1 or 2mm.  Split the parts up, or delete them.  Then it slices.

What does work 95% of the time: only print 1 STL on the plate at a time...  that, is very time consuming.  However, that 5% of the time a single STL doesn't work?  Rotate it and move it over 5mm.  Then it slices!?!

---

I know this comment does little to help solve things.  Instead, can we add more core dump debugging symbols?  In C++, I do believe there is a way to dump the entire heap/stack upon termination.  We can then take that dump file and analyze it (I'm not able to analyze the current dump - pretty novice when it comes to debugging C: mostly golang and higher level languages).

**Comment by supermerill** [comment]
@eduncan911
Understood.
I'll try to output a debug version for linux, to have the stacktrace.

**Comment by eduncan911** [comment]
@supermerill i've got a `Dockerfile` i've built locally to allow me to compile nightly/rc branches.  

when i have some time this weekend, i'll give it a shot.

**Comment by eduncan911** [comment]
> @eduncan911 Understood. I'll try to output a debug version for linux, to have the stacktrace.

FYI, I just resolved all of my crashing issues on Pop_os 22.04 LTS (aka Ubuntu 22.04 LTS w/gnome).

I switched from the gtk2 AppImage to the standard one (I'm guessing gtk3).

I tried both with the previous Sept 2022 release, and the recent 2.5 release.  The exact same 3mf that crashed 100% of the time, now slices 100% just fine without a crash.

**Comment by SyntaxualSugar** [comment]
I've also been having this issue on Manjaro Linux using SuperSlicer 2.4.58.5-5 from the community repository. I had tried a number of things to get it to not crash when slicing my model (resizing it, moving it a bit to the side, changing various settings, etc), none of which worked.

That said, I have found something that seems to work some of the time. I saw an issue for PrusaSlicer, and someone in the comments mentioned that there was a recent fix for an issue that caused crashes when the filename contained mangled international characters. Going off of that, I tried simplifying my filename and it magically worked. In my case it was fume_extractor.stl -> extractor.stl
